src/Lecture3.hs

```diff
@@ -122,19 +133,24 @@ data Reward = Reward
     } deriving (Show, Eq)
 
 instance Semigroup Reward where
+  (<>) (Reward a True) (Reward b _) = Reward (a <> b) True
+  (<>) (Reward a _) (Reward b True) = Reward (a <> b) True
+  (<>) (Reward a _) (Reward b _) = Reward (a <> b) False
 
 
 instance Monoid Reward where
+  mempty = Reward (Gold 0) False
```

Comment:
You can use `mempty` for `Gold` here because it already has a `Monoid` instance 🏅 
```suggestion
  mempty = Reward mempty False
```